src/app.rs

```diff
@@ -0,0 +1,51 @@
+use crate::contract_leaderboard::get_contracts;
+use crate::db::get_db_pool;
+use crate::state::AppState;
+use axum::{routing::get, Router};
+use sqlx::PgPool;
+use std::sync::Arc;
+
+pub async fn get_app() -> Router {
+    let db_pool = get_db_pool().await;
+    get_app_with_db_pool(db_pool)
+}
+
+fn get_app_with_db_pool(db_pool: PgPool) -> Router {
+    let shared_state = Arc::new(AppState { db_pool });
+    tracing_subscriber::fmt::init();
+    Router::new()
+        .route("/contracts", get(get_contracts))
+        .with_state(shared_state)
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::contract_leaderboard::ContractEntry;
+    use axum::{body::Body, http::Request};
+    use tower::ServiceExt;
+
+    #[tokio::test]
+    async fn test_get_contracts_against_dev_db() {
+        let app = get_app().await;
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .uri("/contracts")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .expect("Request Failed");
+        assert_eq!(response.status(), 200);
+        let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
+        let contracts: Vec<ContractEntry> = serde_json::from_slice(&body).unwrap();
+        assert_eq!(contracts.len(), 100);
+
+        for contract in contracts {
+            assert!(contract.address.len() > 0);
+            assert_eq!(&contract.address[..2], "0x");
+            assert!(contract.base_fees > 0.0);
```

Comment:
This one I think works, every contract should have a non-zero f64 amount of base fees, but I do have to ask, what are you really testing for here? If just anything, then whatever haha, test anything. Checking the length of the array of contracts may be quicker. In Rust you can rest assured that if sqlx gives you a list of contracts there will be a list of contracts. Furthermore some industrious engineering by me _should_ mean that you won't encounter weirdness like some contract addresses being invalid but something may have slipped through of course.

Reply:
Yes these tests more or less just test "anything" - i.e. it tests that the general setup works. I.e. the service is connected to the database and can pass through the data. It's really more meant as a starting point to set up a test CI that we can later extend with more meaningful tests.